File: SP-API/Domain/Item.cs
using System.Collections.Generic;

namespace Domain
{
    public class Item
    {
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public string Name { get; set; }
        public string ItemCode { get; set; }
        public string Comment { get; set; }
        public bool IsNew { get; set; }
        public virtual ItemCategory ItemCategory { get; set; }
        public virtual List<ItemBatch> ItemBatches { get; set; }
    }
}

File: SP-Client/src/app/_models/item.ts
export class Item {
  id: number;
  categoryId: number;
  name: string;
  itemCode: string;
  comment: string;
  categoryName: string;
  isNew: boolean;

  constructor(item) {
    this.id = item.id;
    this.name = item.name;
    this.itemCode = item.itemCode;
    this.comment = item.comment;
    this.isNew = item.isNew;
    this.categoryId = item.categoryId;
    this.categoryName = item.categoryName;
  }
}
